pkg/operator2/workload/workload_controller.go

```diff
@@ -0,0 +1,370 @@
+package workload
```

Comment:
this is really super much boilerplate. Wondering how to make that generic and reuse cc @mfojtik